content/en/docs/instrumentation/go/getting-started.md

```diff
@@ -1,681 +1,1427 @@
-created, you are deferring a function to flush and stop it, and registering it
-as the global OpenTelemetry [`TracerProvider`].
-
-Do you remember in the previous instrumentation section when we used the global
-[`TracerProvider`] to get a [`Tracer`]? This last step, registering the
-[`TracerProvider`] globally, is what will connect that instrumentation's
-[`Tracer`] with this [`TracerProvider`]. This pattern, using a global
-[`TracerProvider`], is convenient, but not always appropriate.
-[`TracerProvider`]s can be explicitly passed to instrumentation or inferred from
-a context that contains a span. For this simple example using a global provider
-makes sense, but for more complex or distributed codebases these other ways of
-passing [`TracerProvider`]s may make more sense.
-
-## Putting It All Together
-
-You should now have a working application that produces trace telemetry data!
-Give it a try.
-
-```console
-$ go run .
-What Fibonacci number would you like to know:
-42
-Fibonacci(42) = 267914296
-What Fibonacci number would you like to know:
-^C
-goodbye
-```
+</details>
 
-A new file named `traces.txt` should be created in your working directory. All
-the traces created from running your application should be in there!
+## Custom instrumentation
 
-## Error handling
+Instrumentation libraries captures telemetry at the edges of your systems, such
+as inbound and outbound HTTP requests, but it doesn't capture what's going on in
+your application. For that you'll need to write some custom
+[manual instrumentation](../manual/).
 
-At this point, you have a working application and it is producing tracing
-telemetry data. Unfortunately, there is an error in the core functionality of
-the `fib` module.
+Modify `rolldice.go` to include custom instrumentation using OpenTelemetry API:
 
-```console
-$ go run .
-What Fibonacci number would you like to know:
-100
-Fibonacci(100) = 3736710778780434371
-# …
-```
+```go
+package main
 
-The 100-th Fibonacci number is `354224848179261915075`, not
-`3736710778780434371`! This application is only meant as a demo, but it
-shouldn't return incorrect values. Update the `Fibonacci` function to return an
-error instead of an incorrect value:
+import (
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
 
-```go
-// Fibonacci returns the n-th Fibonacci number. An error is returned if the
-// Fibonacci number cannot be represented as a uint64.
-func Fibonacci(n uint) (uint64, error) {
-	if n <= 1 {
-		return uint64(n), nil
-	}
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
 
-	if n > 93 {
-		return 0, fmt.Errorf("unsupported Fibonacci number %d: too large", n)
-	}
+var (
+	tracer  = otel.Tracer("rolldice")
+	meter   = otel.Meter("rolldice")
+	rollCnt metric.Int64Counter
+)
 
-	var n2, n1 uint64 = 0, 1
-	for i := uint(2); i < n; i++ {
-		n2, n1 = n1, n1+n2
+func init() {
+	var err error
+	rollCnt, err = meter.Int64Counter("roll_counter",
```

Comment:
Since otel seems to prefer dots to underscores, lets use that here.  Also, probably not best to include the type in the name, so maybe `dice.rolls` as the name?

Reply:
I agree. I will also update in other places.

Reply:
Resolved with https://github.com/open-telemetry/opentelemetry.io/pull/3310/commits/f3cf8cc53f4e9d2029521be050cd0bb71c770e7b